src/components/GenericVectorBaseMap/GenericVectorBaseMap.js

```diff
@@ -0,0 +1,82 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { MapContainer, useMap, WMSTileLayer } from 'react-leaflet';
+import 'leaflet/dist/leaflet.css';
+import 'leaflet.vectorgrid';
+import L from 'leaflet';
+import 'proj4';
+import 'proj4leaflet';
+import { projCRSOptions } from '../../utils/crs';
+import VectorGridLayer from './VectorGridLayer';
+
+const GenericVectorBaseMap = ({
+    tileset: { url, projection, tileMatrix },
+    center,
+    zoom,
+    mapRef,
+    vectorTileOptions,
+    wmsUrl,
+    wmsLayerOptions,
```

Comment:
I wonder if the `wms...` props should be given names that indicate their use, something more specific like `wmsLabelLayerUrl`, etc.? Or do you think that the `GenericVectorBaseMap` is/could be a more generalized tool, where the WMS layer would be used for other things?